pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -120,13 +120,18 @@
         category = CATEGORY_SERVER,
         doc = "The port for serving binary protobuf requests"
     )
-
     private Optional<Integer> brokerServicePort = Optional.of(6650);
     @FieldContext(
         category = CATEGORY_SERVER,
         doc = "The port for serving tls secured binary protobuf requests"
     )
     private Optional<Integer> brokerServicePortTls = Optional.empty();
+    @FieldContext(
+            category = CATEGORY_SERVER,
+            doc = "The maximum duration (in seconds) to wait for each broker service network port to close "
+                    + "at broker service shutdown. No waiting will take place when the value is set to <= 0."
+    )
+    private int brokerServicePortCloseTimeoutInSeconds = 2;
```

Comment:
Since there are many executors that we're shutting down, we should just have 1 single configuration. It might even make sense to make it related to the setting: 

```
# Time to wait for broker graceful shutdown. After this time elapses, the process will be killed
brokerShutdownTimeoutMs=60000
```

Reply:
Good suggestion. I'll try to address this tomorrow.

Reply:
@merlimat I'm a bit unsure how to handle this. I created a separate experiment in https://github.com/lhotari/pulsar/commit/2999b2b294066f018a01db01fa77b27eb0914ef4 where there wouldn't be a separate setting for waiting for broker port listeners to shutdown. Would that type of approach make sense?
I didn't find other ways to make the shutdown use the `brokerShutdownTimeoutMs` setting.
I have some concerns of that type of approach since the behavior of shutdown would be different if the port listener shutdown is asynchronous. Is that a valid concern?

Reply:
The final changes in this PR are addressing Matteo's proposal.